packages/shared/src/hooks/useShareOrCopyLink.ts

```diff
@@ -0,0 +1,30 @@
+interface UseShareOrCopyLinkProps {
+  link: string;
+  text: string;
+  trackEvent?: () => void;
+  copyLink?: () => Promise<void>;
+}
+export function useShareOrCopyLink({
+  link,
+  text,
+  trackEvent,
+  copyLink,
+}: UseShareOrCopyLinkProps): () => Promise<void> {
+  const onShareOrCopy = async () => {
+    trackEvent?.();
+    if ('share' in navigator) {
+      try {
+        await navigator.share({
+          text,
+          url: link,
+        });
+      } catch (err) {
+        // Do nothing
+      }
+    } else {
+      copyLink?.();
```

Comment:
I think it's better to have the copy functionality here and then call a callback after the actual copy to show a notification of sorts

Reply:
I think the the difficult part is that some use the post copy hook, which will auto hide the message and some don't
That's why I thought it would be better to leave it up to the implementation to decide what copy function to use.

Reply:
Make sense then. Let's keep the current implementation